pkg/ext/stores/tokens/tokens.go

```diff
@@ -303,6 +361,37 @@ func (t *Store) Delete(
 	return obj, true, nil
 }
 
+// deleteToken is a variant of Delete for use by DeleteCollection. It avoids
+// doing a second `token get` calling on `Delete` would invoke. Delegates to the
+// actual store method after some generic boilerplate.
+func (t *Store) deleteToken(
+	ctx context.Context,
+	secret *corev1.Secret,
+	deleteValidation rest.ValidateObjectFunc,
+	options *metav1.DeleteOptions) (runtime.Object, bool, error) {
```

Comment:
I know it came from kubeconfigs but we can simplify things by returning the concrete object here.
```suggestion
func (t *Store) deleteToken(
	ctx context.Context,
	secret *corev1.Secret,
	deleteValidation rest.ValidateObjectFunc,
	options *metav1.DeleteOptions) (*ext.Token, bool, error) {
```
and remove checks on L323.
I'll need to change this in Kubeconfigs.